includes/RestApi/SiteGenController.php

```diff
@@ -210,45 +191,43 @@ public function generate_sitegen_meta( \WP_REST_Request $request ) {
 	 * @return array
 	 */
 	public function get_homepages( \WP_REST_Request $request ) {
+		$existing_homepages = SiteGenService::get_homepages();
+		if ( ! empty( $existing_homepages ) ) {
+			return new \WP_REST_Response( $existing_homepages, 200 );
+		}
 
 		$site_description = $request->get_param( 'site_description' );
-		$regenerate       = $request->get_param( 'regenerate' );
 		$site_info        = array( 'site_description' => $site_description );
-		// If the option exists and is not empty, return it.
-		$existing_homepages = get_option( Options::get_option_name( 'sitegen_homepages' ), array() );
-		if ( ! empty( $existing_homepages ) && ! $regenerate ) {
-			return new \WP_REST_Response( $existing_homepages, 200 );
-		}
-		$target_audience = SiteGenService::instantiate_site_meta( $site_info, 'target_audience' );
-		$content_style   = SiteGenService::instantiate_site_meta( $site_info, 'content_tones' );
 
-		if ( ! $target_audience || is_wp_error( $target_audience ) ) {
+		$target_audience = SiteGenService::instantiate_site_meta( $site_info, 'target_audience' );
+		if ( is_wp_error( $target_audience ) ) {
 			return new \WP_Error(
 				'nfd_onboarding_error',
 				__( 'Required data is missing.', 'wp-module-onboarding' ),
 				array( 'status' => 400 )
 			);
 		}
-		if ( ! $content_style || is_wp_error( $content_style ) ) {
+
+		$content_style = SiteGenService::instantiate_site_meta( $site_info, 'content_tones' );
+		if ( is_wp_error( $content_style ) ) {
 			return new \WP_Error(
 				'nfd_onboarding_error',
 				__( 'Required data is missing.', 'wp-module-onboarding' ),
 				array( 'status' => 400 )
 			);
```

Comment:
The return value is itself a `wp_error` should we just return that? if either `$target_audience || $content_style` is an `is_wp_error`? 

Also I was wondering if we mention in instantiate_site_meta which key failed as we can never know in this case until we check the db.